### ISSUE TITLE
Add possibility to react on mouseover of a page - for example for prefetching data of the next page

### ISSUE THREAD

**Issue by domvo** [AdeleD/react-paginate#351]
First of all thanks for this package, it helps us tremendously!

We would like to implement a prefetching mechanism when the user hovers over the next/prev button or a certain page link. In my opinion there would be two viable ways to implement this.

1. add some kind of `onPageLinkHover` prop.
2. add the possibility to render a completely custom pageLink

Are there any considerations for such a feature already? I couldn't find a related issue. I'd be willing to contribute for this.

Thanks!

**Comment by AdeleD** [comment]
Thank you!
You can use the `eventListener` prop.

Example:
```javascript
<ReactPaginate 
  pageCount={20}
  eventListener="onMouseOver" />
```

(I'm sorry for the late reply).

**Comment by MonsieurV** [comment]
After consideration, the simpler way would indeed to allow for custom rendering of controls - so all these edge cases could be implemented freely.

Closed in favor of #416
